### PR TITLE
REF: de-duplicate code in liboffsets

### PR DESCRIPTION
**Pull request by jbrockmendel**
```
In [2]: dti = pd.date_range("2016-01-01", periods=10000, freq="S")              
In [3]: off = pd.offsets.BMonthEnd(2)                                           
In [4]: %timeit dti + off                                                       
3.36 ms ± 14 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)  # <-- master
998 µs ± 12.5 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)  # <-- PR
```

### CODE REVIEW

File: pandas/_libs/tslibs/offsets.pyx [hunk 1]
```diff
@@ -4086,15 +3958,15 @@ cdef inline int get_day_of_month(npy_datetimestruct* dts, day_opt) nogil except?
     cdef:
         int days_in_month
 
-    if day_opt == 'start':
+    if day_opt == "start":
```
Comment (jreback) [inline pandas/_libs/tslibs/offsets.pyx:3961]: maybe want to make these an enum

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
ahh guess merge something else, can you rebase again

File: pandas/_libs/tslibs/offsets.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/offsets.pyx:3961]: yah i looked at this, there are a couple of usages that dont follow this pattern yet

**Comment by jbrockmendel** [comment]
rebased+green.  i think just one de-duplication branch after this

**Pull request merged**